components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/client/authentication/PublicClientAuthenticator.java

```diff
@@ -0,0 +1,178 @@
+
+    /**
+     * Returns the execution order of this authenticator.
+     *
+     * @return Execution place within the order.
+     */
+    @Override
+    public int getPriority() {
+
+        return 200;
+    }
+
+    /**
+     * Authenticates the client.
+     *
+     * @param request                 HttpServletRequest which is the incoming request.
+     * @param bodyParams              Body parameter map of the request.
+     * @param oAuthClientAuthnContext OAuth client authentication context.
+     * @return Whether the authentication is successful or not.
+     * @throws OAuthClientAuthnException
+     */
+    @Override
+    public boolean authenticateClient(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
+            oAuthClientAuthnContext) {
+
+        return true;
+    }
+
+    /**
+     * Returns whether the incoming request can be authenticated or not using the given inputs.
+     *
+     * @param request    HttpServletRequest which is the incoming request.
+     * @param bodyParams Body parameters present in the request.
+     * @param context    OAuth2 client authentication context.
+     * @return True if can be authenticated, False otherwise.
+     */
+    @Override
+    public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bodyParams, OAuthClientAuthnContext
+            context) {
+
+        String clientId = getClientId(request, bodyParams, context);
+
+        try {
+            if (isClientIdExistsAsParams(bodyParams)) {
+                if (canBypassClientCredentials(context.getClientId())) {
+                    if (clientId != null) {
+                        context.setClientId(clientId);
+                    }
+                    return true;
+                } else {
+                    log.error("The Application (Service Provider) with client ID : " + clientId +
+                            " has not enabled the option \"Allow authentication without the client secret\" " +
+                            "and no valid Authorization Header exists in the request.");
+                }
+            } else {
+                log.error("Application with the given client ID " + clientId + "is not found");
+            }
+        } catch (InvalidOAuthClientException e) {
+            log.error("Error in retrieving an Application (Service Provider) with client ID : " + clientId, e);
+        } catch (IdentityOAuth2Exception e) {
+            log.error("Error in Application (Service Provider) with client ID : " + clientId, e);
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("The Client ID is not present in the request.");
+        }
+
+        return false;
+    }
+
+    /**
+     * Get the name of the OAuth2 client authenticator.
+     *
+     * @return The name of the OAuth2 client authenticator.
+     */
+    @Override
+    public String getName() {
+
+        return "PublicClientAuthenticator";
```

Comment:
Define the string value as a constant.

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1058